src/terminal/init.lua

```diff
@@ -18,6 +18,26 @@ local M = {
   _DESCRIPTION = "Cross platform terminal library for Lua (Windows/Unix/Mac)",
 }
 
+-- Helper function to resolve negative indices
+local function resolve_index(index, max_value)
+  if index < 0 then
+    return max_value + index + 1
+  end
+  return index
+end
```

Comment:
a `utils` module was added, would that module be a better location for this function?